### ISSUE TITLE
[REVIEW]: PyFstat: a Python package for continuous gravitational-wave data analysis

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3000]
**Submitting author:** @dbkeitel (<a href="http://orcid.org/0000-0002-2824-626X">David Keitel</a>)
**Repository:** <a href="https://github.com/PyFstat/PyFstat/" target ="_blank">https://github.com/PyFstat/PyFstat/</a>
**Version:** v1.11.5
**Editor:** @danielskatz
**Reviewer:** @khanx169, @RobertRosca
**Archive:** <a href="https://doi.org/10.5281/zenodo.4660591" target="_blank">10.5281/zenodo.4660591</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355/status.svg)](https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355"><img src="https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355/status.svg)](https://joss.theoj.org/papers/9a0336ed9451fb0852dfe662d9842355)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/PyFstat/PyFstat/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@khanx169 & @RobertRosca, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @danielskatz know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PyFstat/PyFstat/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@dbkeitel) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/PyFstat/PyFstat/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@dbkeitel) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @khanx169, @RobertRosca it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3000 with the following error: 

 Can't find any papers to compile :-(

**Comment by danielskatz** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
@khanx169 and @RobertRosca - Thanks for agreeing to review this submission.

This is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

Please read the first couple of comments in this issue carefully, so that you can accept the invitation from JOSS and be able to check items, and so that you don't get overwhelmed with notifications from other activities in JOSS.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#3000` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@danielskatz) if you have any questions/concerns.

**Comment by danielskatz** [comment]
@RobertRosca & @khanx169  - it's great to see that you have started your reviews - thanks!

**Comment by dbkeitel** [comment]
Thanks from my side too! There's apparently a numpy-lalsuite version incompatibility at the moment that breaks our automated tests, I'm looking into fixing it by a temporary version pinning.

**Comment by whedon** [comment]
:wave: @RobertRosca, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @khanx169, please update us on how your review is going.

**Comment by danielskatz** [comment]
👋 @khanx169, @RobertRosca - It looks like you all have started, which is great. I just want to check in and see how things are going.  Are you blocked on anything?

**Comment by khanx169** [comment]
> 👋 @khanx169, @RobertRosca - It looks like you all have started, which is great. I just want to check in and see how things are going. Are you blocked on anything?

@danielskatz - thanks for checking in. The last two weeks were incredibly busy which caused a little bit of slowdown. However, I plan to pick up the pace again tomorrow and hopefully finish by this weekend if everything goes smoothly.

**Comment by RobertRosca** [comment]
@danielskatz same on my side as well, been very but I plan on finishing this weekend or by Tuesday next week.

**Comment by dbkeitel** [comment]
Thanks @khanx169 @RobertRosca ! Let us know if you have any questions. We've had two minor releases in the meantime and the dependency issue mentioned above is resolved.

**Comment by khanx169** [comment]
@danielskatz @dbkeitel I've gone through the review checklist and everything looks good to me. I've one minor suggestion: In the Github repo, there's a `run_all_examples.py` script. The docs point to GitHub, but it might be good to also explicitly mention this script there (especially since some examples need to be executed in order).

**Comment by danielskatz** [comment]
Thanks @khanx169! That sounds like a good suggestion.

**Comment by dbkeitel** [comment]
@khanx169 Thanks for the positive feedback!

> I've gone through the review checklist and everything looks good to me. I've one minor suggestion: In the Github repo, there's a run_all_examples.py script. The docs point to GitHub, but it might be good to also explicitly mention this script there (especially since some examples need to be executed in order).

This script was so far mostly intended for our own use - we run it as a manually-triggered action before releases to make sure the examples don't break. (But not automatically, because it takes much longer to run than the regular integration tests.) But if you found it useful, we're of course happy do document it more clearly.

To make sure, where are you suggesting to mention it?
- the README
- readthedocs
- both

**Comment by khanx169** [comment]
> @khanx169 Thanks for the positive feedback!
> 
> > I've gone through the review checklist and everything looks good to me. I've one minor suggestion: In the Github repo, there's a run_all_examples.py script. The docs point to GitHub, but it might be good to also explicitly mention this script there (especially since some examples need to be executed in order).
> 
> This script was so far mostly intended for our own use - we run it as a manually-triggered action before releases to make sure the examples don't break. (But not automatically, because it takes much longer to run than the regular integration tests.) But if you found it useful, we're of course happy do document it more clearly.
> 
> To make sure, where are you suggesting to mention it?
> 
> * the README
> * readthedocs
> * both

@dbkeitel I was thinking specifically of the `Examples` section of readthedocs. I understand most users would probably want to look at some specific example at a time, but my thought process when I reached that section was: "It'd be nice if I could run all the examples first, and then come back and look at the scripts and outputs one by one to familiarize myself with the package". I suspect occasionally some odd user might think like this as well. Currently, the cumulative runtime for all examples is not terrible, but if more examples are added in the future, perhaps interactive notebooks (with the expected output for each cell already displayed) would be more useful.

I liked that the README is concise, but feel free to update both if you think that'd be more appropriate. Thanks.

**Comment by RobertRosca** [comment]
Hey, I also went through the checklist, everything looks good!

I have some suggestions but none of them are major:

- Installation:
  - Pretty subjective, but I think having all the extra requirements under `extras_require` would be nice, then you could have some for `docs`, `style`, `test`, etc... which makes development and contribution a bit easier, also makes the CI scripts cleaner since everything can be a `pip install ".[style,test,...]"` command instead of using separate requirements files
  - The only objective benefit to this is that removing the requirements files makes it easier to maintain as you don't have to keep their pinned versions in sync
  - Also thanks for the quick response to my issue!
- Example usage:
  - Nice examples!
  - This PR https://github.com/PyFstat/PyFstat/pull/226 did work on automatically generating notebooks from the python scripts, I think using ipynb files directly would be nicer for the examples
  - That way when the examples get rendered in the docs they could include some rich output like plots
  - Same goes for the binder integration, currently the conversion from .py to .ipynb makes just one big cell with all the code in it, saving the notebook files directly (or using something like https://github.com/mwouts/jupytext) would let you have richer examples
  - And there could (where it makes sense) be some additional explanations about what's going on
  - This also makes it easier to include the examples as part of the tests directly with https://github.com/computationalmodelling/nbval
- Automated tests:
  - Codecov could be included as part of the CI, ran it locally and the coverage (with just the `tests.py` file) is around 62%
  - Also provides coverage details for PRs, letting you know if a contribution lowers the test coverage
  - Going back to the examples point, if the examples were ipynb files with the output saved in them then nbval could check that the output has not changed, and executing the examples would probably bump the coverage up a decent amount

The article mentions that one of the aims of PyFstat is to make this kind of analysis more accessible to new researchers and students, so I think converting the examples from python scripts to notebooks and then fleshing them out with more text and accompanying plots would be nice.

It's very easy to set up and run the examples yourself but I think that having the output saved and available in the docs makes a bit more beginner friendly. Reading through Asad Khan's suggestions we're thinking along the same lines :smile:

Overall this looks great! To be clear all the suggestions above are very minor, I'm happy with the current state of the submission and think it can be accepted now

**Comment by danielskatz** [comment]
Thanks @khanx169 and @RobertRosca - you both are great!

@RobertRosca - there are a lot of your checkboxes in the review unchecked. Can you check all those you are comfortable with, and if any need more work before you can check them (as opposed to suggestions for improvements that you would like but don't think are required), please let us know about that work too, though I don't think you are saying this.

**Comment by RobertRosca** [comment]
@danielskatz ah yes my bad, forgot to actually check them off as I went through everything, that's fixed now!

**Comment by danielskatz** [comment]
Thanks - so over to @dbkeitel now to make the optional changes and improvements ...

Once this is done, let us all know so we can check how things look and do a final signoff, then we can do the final acceptance and publishing

**Comment by dbkeitel** [comment]
Thanks a lot for the helpful suggestions! It's late here in Spain, so I'll respond in more detail tomorrow.

**Comment by dbkeitel** [comment]
Thanks again to both reviewers! I've now opened issues over on our repo to track the suggestions.

I think we can resolve @khanx169's suggestion about the all-examples script quickly.

For @RobertRosca's suggestions, the examples handling and codecov are something we had planned (but not yet opened issues for) anyway, but the given pointers will be very helpful to it. The suggestion about installation / requirements management also sounds very sensible. It may however take us some time to get around to resolving these three points.

**Comment by RobertRosca** [comment]
Awesome! If I have free time I may do some PRs for those issues myself, so to avoid duplicating work could you (or others working on the project) leave a quick comment on an issue saying that work has started on them already?

**Comment by dbkeitel** [comment]
Hi @RobertRosca that would be awesome and much appreciated!
Generally, if there is no answer to one of the issues yet, nor a linked PR as in https://github.com/PyFstat/PyFstat/issues/287 , then you can safely assume we haven't started on it yet.

**Comment by danielskatz** [comment]
👋 @dbkeitel - this progress looks good, but also be aware that it's up to you if we move to publication now or wait for these optional improvements. I'm happy to wait for them, but we also could publish as is.

**Comment by dbkeitel** [comment]
@danielskatz I was wondering if there's a standard way to link to a package's ASCL entry in the left sidebar of the article, or if the easiest is to include it in the acknowledgments?

**Comment by danielskatz** [comment]
Could you use a reference? I don't think we can use the left column for this. Maybe @arfon will gave another suggestion?

**Comment by arfon** [comment]
> @danielskatz I was wondering if there's a standard way to link to a package's ASCL entry in the left sidebar of the article, or if the easiest is to include it in the acknowledgments?

I'm afraid we can't support custom content in the left-hand column but you're welcome to add it as an explicit callout in the text (i.e., an acknowledgment) or cite directly as @danielskatz suggests.

**Comment by dbkeitel** [comment]
Thanks both for checking, I'll just add it as suggested.

Also we'll be merging a few small infrastructure fixes/improvements soon (including codecov integration as suggested by @RobertRosca ), then make another release, and then would be ready for publication.

**Comment by dbkeitel** [comment]
@danielskatz : We've made a new [v1.11.4 release](https://github.com/PyFstat/PyFstat/releases/tag/v1.11.4) of the package including the following behind-the-scenes improvements, including reviewer comments:

- python 3.9 now supported
- improvements to documentation and examples
- for developers: flake8-docstrings and flake8-executable rules now enforced
- added a codemeta.json file
- started tracking test coverage with codecov

We'll continue improving documentation, examples and test coverage in the future, but for now we would be glad to proceed with publishing the paper. Two small final issues from our side, sorry for being difficult "customers" and thanks for the great experience so far:

1. I noticed a small inconsistency in citation formats:
 * papers with medium-length, explicitly given author lists become `et al.` in the compiled pdf, for example one with bibtex entry like `author = "Keitel, David and Prix, Reinhard and Papa, Maria Alessandra and Leaci, Paola and Siddiqi, Maham",`.
 * but for collaboration papers where inspire only puts e.g. `author = "Acernese, F. and others",`, these are displayed as `& others` in the pdf.
My (limited) understanding is that the bibtex+LaTeX combo normally knows that "and others" has to be translated to "et al.", but pandoc apparently doesn't. I tried a few things (just writing `et al.` in the bibtex author field -> disappears in the pdf; pasting full author lists (1000+ for some) -> works but compilation becomes extremely slow) but is there a recommended trick?

2. We'll also need to do a final update of our funding acknowledgments before "freezing" the paper.

**Comment by dbkeitel** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
I'll see if I can find an answer to 1, and then we can do 2.  If we can't fix 1, I don't think it's a big problem though.

**Comment by danielskatz** [comment]
When we are ready, the next steps will be
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

**Comment by arfon** [comment]
> My (limited) understanding is that the bibtex+LaTeX combo normally knows that "and others" has to be translated to "et al.", but pandoc apparently doesn't. I tried a few things (just writing et al. in the bibtex author field -> disappears in the pdf; pasting full author lists (1000+ for some) -> works but compilation becomes extremely slow) but is there a recommended trick?

@tarleb - any suggestions here?

**Comment by dbkeitel** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: :point_left:

**Comment by dbkeitel** [comment]
- citation formats: Now I've simply resorted to entering the first 3 authors + "and others" in the `.bib` file, the result is the correct "first tauthor et al." in the pdf text. Of course the bibliography entries at the end themselves now have those 3 + "& others"; if that's ok with JOSS style, this solution is good for me.
- acknowledgments: also updated
- deposited version: we're automatically depositing our releases on Zenodo, but for the one I did yesterday there has been an issue with the PyPI/Conda uploads, which I'll try to fix and then make another proper "reference version" for the paper, and will post the corresponding Zenodo DOI here.

**Comment by dbkeitel** [comment]
@danielskatz v1.11.5 of the package, with DOI [10.5281/zenodo.4659766](https://doi.org/10.5281/zenodo.4659766), is ready as a reference version for the paper.

However, it doesn't have the longer title of the paper, and the order of the author list is different (original author first). Should I make a separate manual deposit off the `joss-paper` branch with exactly matching metadata, or is this one fine?

**Comment by dbkeitel** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03000/joss.03000/10.21105.joss.03000.pdf) :page_facing_up: :point_left:

**Comment by danielskatz** [comment]
@dbkeitel - you should be able to go to the Zenodo page and edit the metadata

**Comment by dbkeitel** [comment]
> @dbkeitel - you should be able to go to the Zenodo page and edit the metadata

Of course, but then it would look odd that this one version has different data than all others before and after. If JOSS needs one deposit that exactly matches the paper, then I'd prefer to do a "duplicate", if that's ok.

**Comment by danielskatz** [comment]
> Of course, but then it would look odd that this one version has different data than all others before and after. 

As Emerson said, "A foolish consistency is the hobgoblin of little minds" - but it's up to you :)

> If JOSS needs one deposit that exactly matches the paper, then I'd prefer to do a "duplicate", if that's ok.

Sure, that's fine.

**Comment by dbkeitel** [comment]
@danielskatz Here's the reference deposit: https://doi.org/10.5281/zenodo.4660591

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.4660591 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4660591" target="_blank">10.5281/zenodo.4660591</a> is the archive.

**Comment by danielskatz** [comment]
@whedon set v1.11.5 as version

**Comment by whedon** [comment]
OK. v1.11.5 is the version.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3000 with the following error: 

 Can't find any papers to compile :-(

**Comment by danielskatz** [comment]
@whedon accept from branch joss

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.7935/GT1W-FZ16 is OK
- 10.1103/PhysRevD.58.063001 is OK
- 10.1007/978-3-540-76965-1_24 is OK
- 10.1016/j.parco.2011.09.001 is OK
- 10.1103/PhysRevD.89.064023 is OK
- 10.1088/0264-9381/32/2/024001 is OK
- 10.1088/0264-9381/32/7/074001 is OK
- 10.1093/mnras/stv2422 is OK
- 10.1142/S021773231730035X is OK
- 10.1103/PhysRevD.97.103020 is OK
- 10.1103/PhysRevD.98.063011 is OK
- 10.1088/1361-6382/aade34 is OK
- 10.3390/universe5110217 is OK
- 10.1103/PhysRevLett.124.191102 is OK
- 10.1093/mnras/staa3624 is OK
- 10.1016/j.softx.2020.100634 is OK
- 10.1103/PhysRevD.103.064017 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2181

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2181, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss 
```

**Comment by danielskatz** [comment]
@dbkeitel - can you confirm this is ready to go from your point of view?  I will also be doing a final proof read...

**Comment by danielskatz** [comment]
My proof read has found some minor changes that are needed - see https://github.com/PyFstat/PyFstat/pull/304.  Please merge this, or let me know what you disagree with.

**Comment by danielskatz** [comment]
@whedon accept from branch joss

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.7935/GT1W-FZ16 is OK
- 10.1103/PhysRevD.58.063001 is OK
- 10.1007/978-3-540-76965-1_24 is OK
- 10.1016/j.parco.2011.09.001 is OK
- 10.1103/PhysRevD.89.064023 is OK
- 10.1088/0264-9381/32/2/024001 is OK
- 10.1088/0264-9381/32/7/074001 is OK
- 10.1093/mnras/stv2422 is OK
- 10.1142/S021773231730035X is OK
- 10.1103/PhysRevD.97.103020 is OK
- 10.1103/PhysRevD.98.063011 is OK
- 10.1088/1361-6382/aade34 is OK
- 10.3390/universe5110217 is OK
- 10.1103/PhysRevLett.124.191102 is OK
- 10.1093/mnras/staa3624 is OK
- 10.1016/j.softx.2020.100634 is OK
- 10.1103/PhysRevD.103.064017 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2198

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2198, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true from branch joss 
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true from branch joss

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1379527801175470084) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2199
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.03000](https://doi.org/10.21105/joss.03000)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by dbkeitel** [comment]
@danielskatz Thanks for your proofread! I've merged it, then found one more missing comma in the acknowledgments, fixed on our branch just now. But if it's too late for that one now, it's also ok.

**Comment by danielskatz** [comment]
unless it's essential, let's ignore it at this point

**Comment by dbkeitel** [comment]
Ok, thanks for the support throughout!

**Comment by danielskatz** [comment]
Congratulations to @dbkeitel (David Keitel) and co-authors!!

and thanks to @khanx169 and @RobertRosca for reviewing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03000/status.svg)](https://doi.org/10.21105/joss.03000)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03000">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03000/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03000/status.svg
   :target: https://doi.org/10.21105/joss.03000
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03000/status.svg)](https://doi.org/10.21105/joss.03000)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
